aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -130,6 +135,88 @@ Tensor& addmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& ma
   return result;
 }
 
+Tensor& baddmm_out_cuda_impl(Tensor& result, const Tensor& self, const Tensor& batch1, const Tensor& batch2, Scalar beta, Scalar alpha) {
+  TORCH_CHECK(self.dim() == 3, "self must be a 3D tensor");
+  TORCH_CHECK(batch1.dim() == 3, "batch1 must be a 3D tensor");
+  TORCH_CHECK(batch2.dim() == 3, "batch2 must be a 3D tensor");
+
+  IntArrayRef batch1_sizes = batch1.sizes();
+  IntArrayRef batch2_sizes = batch2.sizes();
+  IntArrayRef self_sizes = self.sizes();
+
+  TORCH_CHECK(self_sizes[0] == batch1_sizes[0], "self dim 0 must match batch1 dim 0");
+  TORCH_CHECK(self_sizes[0] == batch2_sizes[0], "self dim 0 must match batch2 dim 0");
+  TORCH_CHECK(self_sizes[1] == batch1_sizes[1], "self dim 1 must match batch1 dim 1");
+  TORCH_CHECK(self_sizes[2] == batch2_sizes[2], "self dim 2 must match batch2 dim 2");
+  TORCH_CHECK(batch1_sizes[2] == batch2_sizes[1], "batch1 dim 2 must match batch2 dim 1");
+
+  if (!result.is_same(self)) {
+    at::native::resize_as_(result, self);
+    if ((beta.isComplex() && beta.to<c10::complex<double>>() != 0.0) ||
+        (beta.to<double>() != 0.0)) {
+      at::native::copy_(result, self);
+    }
+  }
+
+  bool transpose_result = false;
+  bool transpose_batch1, transpose_batch2;
+  int64_t lda, ldb, ldc;
+  Tensor result_;
+  IntArrayRef result_strides = result.strides();
+  IntArrayRef result_sizes = result.sizes();
+
+  if ((result_strides[1] == 1) &&
+      ((result_sizes[2] == 1) || (result_strides[2] >= std::max<int64_t>(1, result_sizes[0])))) {
+    result_ = result;
+  } else if ((result_strides[2] == 1) &&
+    (result_sizes[1] == 1 || (result_strides[1] >= std::max<int64_t>(1, result_sizes[2])))) {
+    transpose_result = true;
+    result_ = result;
+  } else {
+    auto result_ = result.transpose(1, 2).clone();
```

Comment:
The default memory format is contiguous so I don't think it's necessary.

Reply:
default is PreserveMemoryFormat, so it may or may not be contiguous, e.g.
```
In [8]: a=torch.randn(2,3).t()                                                                                                                                                                                        

In [9]: a.stride()                                                                                                                                                                                                    
Out[9]: (1, 3)

In [10]: b=a.clone()                                                                                                                                                                                                  

In [11]: b.stride()                                                                                                                                                                                                   
Out[11]: (1, 3) #it's not contiguous!
```